packages/transport-webrtc/src/private-to-public/listener.ts

```diff
@@ -171,6 +171,37 @@ export class WebRTCDirectListener extends TypedEventEmitter<ListenerEvents> impl
     }
   }
 
+  private async createAndSetCertificate (): Promise<void> {
+    const keyPair = await crypto.subtle.generateKey({
+      name: 'ECDSA',
+      namedCurve: 'P-256'
+    }, true, ['sign', 'verify'])
+
+    const certificate = await generateTransportCertificate(keyPair, {
+      days: this.init.certificateDuration ?? 365 * 10
+    })
+
+    this.certificate = certificate
+    this.setCertificateExpiryTimeout()
+  }
+
+  private setCertificateExpiryTimeout (): void {
+    if (this.certificate == null) {
+      return
+    }
+
+    const expiryDate = new Date(this.certificate.notAfter)
+    const now = new Date()
+    const timeToExpiry = expiryDate.getTime() - now.getTime()
```

Comment:
No need to allocate a new `Date` object:

```suggestion
    const timeToExpiry = expiryDate.getTime() - Date.now()
```